src/sentry/features/manager.py

```diff
@@ -138,24 +138,41 @@ class FeatureManager(RegisteredFeatureManager):
     def __init__(self) -> None:
         super().__init__()
         self._feature_registry: MutableMapping[str, type[Feature]] = {}
+        # Deprecated: Remove entity_features once flagr has been removed.
         self.entity_features: MutableSet[str] = set()
+        self.exposed_features: MutableSet[str] = set()
         self.option_features: MutableSet[str] = set()
         self.flagpole_features: MutableSet[str] = set()
         self._entity_handler: FeatureHandler | None = None
 
-    def all(self, feature_type: type[Feature] = Feature) -> Mapping[str, type[Feature]]:
+    def all(
+        self, feature_type: type[Feature] = Feature, api_expose: bool | None = None
+    ) -> Mapping[str, type[Feature]]:
         """
         Get a mapping of feature name -> feature class, optionally specific to a
         particular feature type.
+
+        :param feature_type: The feature class you want to filter by. eg. (OrganizationFeature | ProjectFeature | SystemFeature)
+        :param api_expose: Only include features that were registered with `api_expose`.
         """
-        return {k: v for k, v in self._feature_registry.items() if issubclass(v, feature_type)}
+        output: dict[str, type[Feature]] = {}
+        for name, feature in self._feature_registry.items():
+            match = False
+            if issubclass(feature, feature_type):
+                match = True
+            if match and api_expose and name not in self.exposed_features:
+                match = False
+            if match:
+                output[name] = feature
```

Comment:
Nit: `{k: v for k, v in self._feature_registry.items() if issubclass(v, feature_type) and (not api_expose or name in self.exposed_features)}`?